packages/tabs/src/Tabs.tsx

```diff
@@ -0,0 +1,285 @@
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import Tab, { TabProps } from './Tab';
+import TabTitle from './TabTitle';
+
+const listStyle = css`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  list-style: none;
+  padding: 0px;
+  margin: 20px 0px 5px 0;
+`;
+
+const listTitle = css`
+  display: inline-block;
+  background-color: ${uiColors.white};
+  color: ${uiColors.gray.base};
+  font-weight: bold;
+  font-size: 16px;
+  line-height: 1;
+  text-transform: capitalize;
+  padding: 3px 24px;
+  cursor: pointer;
+  transition: 150ms color ease-in-out;
+
+  &:focus {
+    color: ${uiColors.blue.base};
+    outline: none;
+  }
+
+  &:hover:not(:focus) {
+    color: ${uiColors.gray.dark3};
+  }
+`;
+
+const activeStyle = css`
+  color: ${uiColors.gray.dark3};
+  cursor: default;
+`;
+
+const disabledStyle = css`
+  pointer-events: none;
+  color: ${uiColors.gray.light2};
+`;
+
+const grayBorder = css`
+  height: 3px;
+  background-color: ${uiColors.gray.light2};
+  position: relative;
+`;
+
+const greenBorder = css`
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  background-color: ${uiColors.green.base};
+  transition: 150ms transform ease-in-out, 150ms width ease-in-out;
+`;
+
+const focusedStyle = css`
+  background-color: ${uiColors.blue.base};
+`;
+
+interface TabsProps {
+  /**
+   * Conent that will appear inside of Tabs component. Should be comprised of at least two Tabs.
+   */
+  children: Array<React.ReactElement>;
+
+  /**
+   * Callback to be executed when a tab is selected.
+   */
+  onChange?: React.ReactEventHandler;
+
+  /**
+   * Tab that should appear active. If value passed to selected prop, component will be controlled by consumer.
+   */
+  selected?: string | null;
+
+  /**
+   * className supplied to Tabs container.
+   */
+  className?: string;
+}
+
+/**
+ * # Tabs
+ *
+ * Tabs component
+ *
+ * ```
+<Tabs onChange={() => execute callback onChange}>
+  <Tab value='tab-1' title='First Tab'>Tab 1</Tab>
+  <Tab value='tab-2' title='Second Tab'>Tab 2</Tab>
+</Tabs>
+```
+ * @param props.children Content to appear inside of Tabs component.
+ * @param props.onChange Callback to be executed when a Tab is selected.
+ * @param props.selected Tab that should appear active. If value passed, component will be controlled by consumer.
+ * @param props.className Classname applied to Tabs container.
+ */
+function Tabs({ children, onChange, selected, className, ...rest }: TabsProps) {
+  const [activeTab, setActiveTab] = useState();
+  const [currentIndex, setCurrentIndex] = useState();
+  const [focusedState, setFocusedState] = useState([] as Array<string>);
+  const listRef = useRef<HTMLUListElement>(null);
+
+  useEffect(() => {
+    const currIdx = React.Children.map(
```

Comment:
Could we spell out `currentIndex` instead?

Reply:
👍